### ISSUE TITLE
Tree operations become slow in very big files

### ISSUE THREAD

**Issue by bakbuk** [giuspen/cherrytree#269]
I have a 15mb file with about 3000 rich text nodes in it. Two operations which were previously very fast had become slow: file save (1-2 seconds), and more crucially, adding a node in the tree (5-6 seconds). Is this sort of slowdown inevitable in such big files or perhaps some optimization is possible to speed things up?

I could provide such big file for testing if that'd be helpful.

**Comment by giuspen** [comment]
I'm in the process of porting to c++ / gtkmm3 as you can read in the forum, after that I'm sure the app will be way faster

**Comment by jkathir** [comment]
Thanks a lot and will keeping watching on the forthcoming fixes of gtkmm3..Have a nice day.

**Comment by jkathir** [comment]
Does it will create index and search through the index like Apache lucene or solr or elastic search ? Searching the contents every time will be a performance issue always until they is a mechanism to build the index and search through it like Acrobat DC creates index, Docfetcher creates index before performs the search.

**Comment by jkathir** [comment]
@giuspen Is porting to c++ / gtkmm3 has been done ? Do you have any updates on this issue?

**Comment by giuspen** [comment]
I'm still working on the porting but there is still a lot of work to do. There may be more minor pygtk2 releases before the first gtkmm3 version will be issued

**Comment by jkathir** [comment]
Is porting to c++ / gtkmm3 is done and the performancee has been fixed?

**Comment by giuspen** [comment]
It is ongoing in the subfolder "future" but still slowly, hope to speed up soon

**Comment by chefarov** [comment]
I think it's mostly proportional to the size of the database itself. The size of my database is 27MB and I need to wait 10-15 seconds to perform a save. My tree consists of:
* 4 top-level nodes
* Max-Depth: 3
* total nodes: 100-130, rich text nodes with pictures (screenshots)

Porting to C++ will be a life-saver. Thank you for the app anyway. Its very useful!

**Comment by giuspen** [comment]
@chefarov for such large db you should use sqlite / ctb - that should not take so long to save

**Comment by chefarov** [comment]
@giuspen You are right of course. I can't believe I had missed that option at the time. Thanks.

**Comment by samuelkathiresan** [comment]
@giuspen is the porting has been completed in the new version?
